### PR TITLE
Add OrderAssignee and related endpoints

### PR DESCRIPTION
**Pull request by marcofucci**
This:
- adds a new `OrderAssignee` model
- adds 2 new endpoints:

`GET /v3/omis/order/<order-id>/assignee`

Returns the list of advisers assigned to an order.

`PATCH /v3/omis/order/<order-id>/assignee`

(Partially) updates the advisers assigned to an order.
By default it keeps the assignees not included in the request body but you can pass the query string `force-delete=1` and it will delete them.

There's a bit of noise because I split the `test_views` into different modules.

[Docs](https://documenter.getpostman.com/view/2403194/omis/6fZxPC3#454e3dd8-3bae-0e73-20b9-0742580a1515)

### CODE REVIEW

File: datahub/metadata/test/factories.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ class Meta:  # noqa: D101
 class TeamRoleFactory(factory.django.DjangoModelFactory):
     """TeamRole factory."""
 
-    id = factory.LazyFunction(lambda: str(uuid.uuid4()))
+    id = factory.LazyFunction(lambda: uuid.uuid4())
     name = factory.Sequence(lambda n: f'name {n}')
```
Comment (reupen) [inline datahub/metadata/test/factories.py]: How about `id = factory.LazyFunction(uuid.uuid4)`?

File: datahub/omis/order/models.py [hunk 2]
```diff
@@ -98,6 +98,62 @@ def __str__(self):
         return f'{self.order} – {self.adviser}'
 
     class Meta:  # noqa: D101
+        ordering = ['created_on']
```
Comment (reupen) [inline datahub/omis/order/models.py]: Do we need an index on created_on?

File: datahub/metadata/test/factories.py (hunk 1 above)
Comment (marcofucci) [inline datahub/metadata/test/factories.py]: very nice

File: datahub/omis/order/models.py (hunk 2 above)
Comment (marcofucci) [inline datahub/omis/order/models.py]: probably

File: datahub/metadata/test/factories.py (hunk 1 above)
Comment (marcofucci) [inline datahub/metadata/test/factories.py]: I might do it in a different PR if you don't mind, I would like to get rid of the `str` everywhere if I manage to.

File: datahub/omis/order/models.py (hunk 2 above)
Comment (marcofucci) [inline datahub/omis/order/models.py]: @reupen I've added an index to the BaseModel and therefore all the subclasses. See if it makes sense.

File: datahub/metadata/test/factories.py (hunk 1 above)
Comment (reupen) [inline datahub/metadata/test/factories.py]: Yeah not everywhere, but the ones being updated already in this PR could be updated...

File: datahub/omis/order/models.py (hunk 2 above)
Comment (reupen) [inline datahub/omis/order/models.py]: Hmm, it's an interesting question about whether we want it on all subclasses of BaseModel. Could be argued either way, so I think it's fine. If we do want it on just a couple of the models though, we should be able to do it by setting `indexes` in the Meta class...

Comment (marcofucci) [inline datahub/omis/order/models.py]: yeah, I think that in general you might want to order by `created_on` quite often as it's a very useful field to base your ordering logic on but that might not be always be the case.

Do you think that having it always on could cause any problems? If not, I would probably leave it everywhere.

File: datahub/omis/order/models.py [hunk 3]
```diff
@@ -98,6 +98,62 @@ def __str__(self):
...
+        future-proofing reasons, that is, if an adviser moves to another team in the future
+        we don't want to change history.
+        """
+        if not self._state.adding and self.__adviser != self.adviser:
+            raise ValueError('Updating the value of adviser isn\'t allowed.')
+
+        if self._state.adding:
+            self.team = self.adviser.dit_team
+            if self.team:
+                self.country = self.team.country
```
Comment (reupen) [inline datahub/omis/order/models.py]: Should it set `self.country` to None if the adviser doesn't have a team?

Comment (marcofucci) [inline datahub/omis/order/models.py]: Wouldn't it be None already by default? `country` is set only when adding and when not None and it never change again.

**Pull request merged**